editor.planx.uk/src/@planx/components/PlanningConstraints/Public.tsx

```diff
@@ -42,19 +43,24 @@ export type InaccurateConstraints =
 export default Component;
 
 function Component(props: Props) {
-  const [inaccurateConstraints, setInaccurateConstraints] =
-    useState<InaccurateConstraints>();
-
-  const siteBoundary = useStore(
-    (state) => state.computePassport().data?.["property.boundary.site"],
+  const [currentCardId, cachedBreadcrumbs, teamSlug, siteBoundary, { x, y, longitude, latitude, usrn }] = useStore(
+    (state) => [
+      state.currentCard?.id,
+      state.cachedBreadcrumbs,
+      state.teamSlug,
+      state.computePassport().data?.["property.boundary.site"],
+      (state.computePassport().data?.["_address"] as SiteAddress) || {},
+    ],
   );
-  const { x, y, longitude, latitude, usrn } =
-    (useStore(
-      (state) => state.computePassport().data?._address,
-    ) as SiteAddress) || {};
+
+  // PlanningConstraints must come after at least a FindProperty in the graph
   const showGraphError = !x || !y || !longitude || !latitude;
 
-  const teamSlug = useStore((state) => state.teamSlug);
+  // Even though this component will fetch fresh GIS data when coming "back",
+  //   still prepopulate any previously marked inaccurateConstraints
```

Comment:
I think we could theoretically populate this component when coming "back" using something like `cachedBreadcrumbs?.[currentCardId]?.["data"]?.["_constraints"]` which is basically the full collated GIS API response.

But in my opinion, while the load is a bit slow, it's in our best interest to always check for the latest data! Feels like a much easier mental model to communicate to users, and a safer guarantee that we're giving the best suggestion we can.

Currently, we do disable `revalidateOnFocus` and SWR has other similar options (https://swr.vercel.app/docs/revalidation#disable-automatic-revalidations) - but again I'm not sure it's totally worth it unless we hear otherwise in feedback ! If we go the full immutable route, which does make for a totally smooth, refresh-free "back" on quick try, we're still breaking that mental model above.

Hopeful that a forthcoming update to the Planning Data API to omit geometry fields when fetching will marginally speed things up, and could also revisit roads query just to ensure we aren't missing an optimisation there (but last I remember it's just a pretty big slow string match).

Reply:
Yep this makes sense, I think the planning data change should have a decent impact here 👍